### PR TITLE
Feature: Invoke Terminal Gif

### PR DESCRIPTION
**Pull request by JustinGrote**
**NOTE: PR is dependent on #44 and must be rebased after #44 is merged with `git rebase --onto master 531d88aa555fe5ba2561fb605d9542f7d68318ea`** 

Adds a cmdlet to "run" a gif in the terminal by temporarily toggling the background setting. It does this in a ThreadJob so as not to block.

For WinPS a "soft" dependency on ThreadJob module is defined as not all users will need this functionality. For PS Core, ThreadJob is built in. Technically we should compare 6.1 and below but 6.1 is end of life now so it's a non-issue.

### CODE REVIEW

File: src/ExportedFunctions/Invoke-MSTerminalGif.ps1 [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+    PS C:\> Invoke-TerminalGif (Search-Giphy -DirectUrl)
+    Triggers a gif in the current Windows Terminal
+.NOTES
+    Requires Windows Terminal Preview
+#>
+    [CmdletBinding()]
+    param (
+        #The URI of the GIF you want to display
+        [Parameter(Mandatory)][uri]$Uri,
+        #The name or GUID of the Windows Terminal Profile in which to play the Gif. If not specified, attempts to intelligently determine it from the executable name or if $ENV:WT_PROFILE was specified by adding -noexit -command \". {$env:WT_PROFILE='myprofilename'}\" to your terminal profile definition.
```
Comment (gpduck) [inline src/ExportedFunctions/Invoke-MSTerminalGif.ps1]: I don't see where it checks $ENV:WT_PROFILE

File: src/ExportedFunctions/Invoke-MSTerminalGif.ps1 [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        } else {
+            Remove-Job $InvokeTerminalGifJob
+        }
+    }
+
+    $TerminalProfile = if ($Name -as [Guid]) {
+        Get-MSTerminalProfile -Guid $Name -ErrorAction stop
+    } else {
+        Get-MSTerminalProfile -Name $Name -ErrorAction stop
+    }
```
Comment (gpduck) [inline src/ExportedFunctions/Invoke-MSTerminalGif.ps1:51]: Should this throw an error if the profile wasn't found?

Comment (gpduck) [reply src/ExportedFunctions/Invoke-MSTerminalGif.ps1:51]: Nevermind, I see it down there :)

File: src/ExportedFunctions/Invoke-MSTerminalGif.ps1 (hunk 1 above)
Comment (JustinGrote) [reply src/ExportedFunctions/Invoke-MSTerminalGif.ps1]: That is in the next pull request that will automatically detect "this" profile if the command is run inside windows terminal, I will fix it for clarification.

File: src/ExportedFunctions/Invoke-MSTerminalGif.ps1 [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+    #Prepare arguments for the threadjob
+    $TerminalGifJobParams = @{ }
+    ('terminalprofile','uri', 'maxduration', 'stretchmode','acrylic','backgroundimageopacity').foreach{
+        $TerminalGifJobParams.$PSItem = (Get-Variable $PSItem).value
+    }
+
+    $TerminalGifJobParams.modulePath = (Get-Module msterminalsettings | % path) -replace 'psm1$','psd1'
```
Comment (gpduck) [inline src/ExportedFunctions/Invoke-MSTerminalGif.ps1]: Can you update this to `(Get-Module msterminalsettings).path` (or ForEach-Object) so script analyzer won't complain?

Comment (JustinGrote) [reply src/ExportedFunctions/Invoke-MSTerminalGif.ps1]: Sure, this was probably a holdover when I was just hacking away at a prototype.
Fixed https://github.com/gpduck/MSTerminalSettings/pull/45/commits/628720f02b45794e50e032c823fa4bc31e8a510d

**Pull request merged**